microdata_tools/_decrypt.py

```diff
@@ -93,26 +116,52 @@ def untar_encrypted_dataset(input_file: Path, dataset_name: str, untar_dir: Path
 
 
 def _validate_tar_contents(files: List[str], dataset_name: str) -> None:
-    if len(files) == 1:
-        if f"{dataset_name}.json" not in files:
-            raise InvalidTarFileContents(f"{dataset_name}.json not in .tar file")
-        else:
-            return
-
-    if len(files) == 3:
-        if (
-            f"{dataset_name}.json" not in files
-            or f"{dataset_name}.csv.encr" not in files
-            or f"{dataset_name}.symkey.encr" not in files
-        ):
+    if f"{dataset_name}.json" not in files:
+        raise InvalidTarFileContents(f"{dataset_name}.json not in .tar file")
+
+    if len(files) > 1:
+        if f"{dataset_name}.symkey.encr" not in files:
             raise InvalidTarFileContents(
-                f"Tar file for {dataset_name} does not contain the "
-                f"required {dataset_name}.csv.encr "
-                f"or {dataset_name}.symkey.encr file "
-                f"or {dataset_name}.json file "
+                f"Tar file for {dataset_name} does not contain the required "
+                f"{dataset_name}.symkey.encr file"
             )
-    else:
-        raise InvalidTarFileContents(
-            f"Tar file for {dataset_name} contains "
-            f"incorrect number ({len(files)}) of files: {files}"
-        )
+
+        chunk_files = [str for str in files if str.endswith(".csv.encr")]
+
+        if len(chunk_files) == 0:
+            raise InvalidTarFileContents(
+                f"Tar file for {dataset_name} does not contain any chunks files"
+            )
+
+
+def _combine_csv_files(input_dir: Path, output_file: Path) -> None:
+    sorted_chunkpaths = _get_sorted_file_names(input_dir)
+    logger.debug(f"\nCombining {len(sorted_chunkpaths)} files into {output_file}")
+
+    combined_file = open(output_file, "wb")
+    for chunk_number, file_name in sorted_chunkpaths.items():
+        chunkpath = input_dir / file_name
+        with open(chunkpath, "rb") as chunk_file:
+            chunk_data = chunk_file.read()
+            combined_file.write(chunk_data)
+    combined_file.close()
+
+
+# Turn files_names in input dir into a dictionary with the chunk number as key
+# Then return the sorted dictionary
+def _get_sorted_file_names(input_dir: Path) -> dict:
```

Comment:
considering the comment under about the naming of files we could rewrite this function:
```py
def _get_sorted_file_names(directory: Path) -> List[Tuple]:
    try:
        return sorted([
            (int(f[6:], f) for f in os.listdir(directory)
            if f.startswith("chunk_"))
        ])
    except ValueError as e:
        raise SomeError("") from e
```

The reason this should be a handled exception too is we have generated this directory with our code. If there is a file without a number, this should be raised as a fatal error with the data rather than logged to debug only. 👍🏻 Then executor can handle that error by shutting down the pipeline because of faulty data input.